src/sage/matrix/matrix1.pyx

```diff
@@ -733,15 +732,70 @@ cdef class Matrix(Matrix0):
             dtype('int64')  # 64-bit
             sage: b.shape
             (3, 4)
+
+        TESTS::
+
+            sage: # needs numpy
+            sage: matrix(3, range(12)).numpy(copy=False)
+            doctest:warning...
+            DeprecationWarning: passing copy argument to numpy() is deprecated
+            See https://github.com/sagemath/sage/issues/39152 for details.
+            array([[ 0,  1,  2,  3],
+                   [ 4,  5,  6,  7],
+                   [ 8,  9, 10, 11]])
         """
+        if copy is not _MISSING:
+            from sage.misc.superseded import deprecation
+            deprecation(39152, "passing copy argument to numpy() is deprecated")
```

Comment:
I decide to deprecate this feature because:

* `copy=False` was never supported in the first place (it always copies)
* `copy=False` implicitly copies is incompatible with numpy 2.0 interface where `np.array(..., copy=False)` raises `ValueError` if a copy is made
* `copy=*` doesn't work in numpy-based matrices (even after  https://github.com/sagemath/sage/pull/38683 )